src/Navigation.tsx

```diff
@@ -464,19 +470,25 @@ const LINKING = {
 
 function RoutesContainer({children}: React.PropsWithChildren<{}>) {
```

Comment:
I did it here because `Shell` component is forked and `App` is also forked but the logic is the same. Also, this gives us the precise point where extra logic doesn't block the first render so it's a good consolidation point.